### DISCUSSION TITLE
passing git args to filter output failed?

### DISCUSSION THREAD

**Discussion by butterl** [gitpython-developers/GitPython#1175] [Q&A] [answered]
Is it possiable to passing args like `refs/tags/TAG01 ..HEAD --no-merges`  to filter the out put commit ? 

```
repo = Repo(path)
for commit in repo.iter_commits(rev='refs/tags/TAG1.0..HEAD --no-merges'))
    print(commit)
```

the error show like this 
```
  File "/home/ubuntu/.local/lib/python3.8/site-packages/git/objects/commit.py", line 277, in _iter_from_process_or_stream
    finalize_process(proc_or_stream)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/git/util.py", line 329, in finalize_process
    proc.wait(**kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/git/cmd.py", line 408, in wait
    raise GitCommandError(self.args, status, errstr)
git.exc.GitCommandError: Cmd('git') failed due to: exit code(128)
  cmdline: git rev-list refs/tags/TAG01..HEAD --no-merges --
  stderr: 'fatal: bad revision 'refs/tags/TAG01..HEAD --no-merges'
```

the cmd ` git rev-list refs/tags/TAG01..HEAD --no-merges --`  seems work fine in cmdline run

**Comment by Byron** [comment, marked as answer]
I think what you are trying to do might work [with the help of the docs on 'how to use git directly'](https://gitpython.readthedocs.io/en/stable/tutorial.html#using-git-directly) along with the API docs on [iter_commits()](https://gitpython.readthedocs.io/en/stable/reference.html?highlight=iter_commits#git.repo.base.Repo.iter_commits).

With that, you should be able to figure out which options to pass to `git rev-list` as `kwargs` of `iter_commits(…)` to get the desired result.

Even though I close this issue, feel free to post follow ups in the comments.
